src/NLog/Layouts/Layout.cs

```diff
@@ -63,7 +63,7 @@ public abstract class Layout : ISupportsInitialize, IRenderable
         /// 
         /// Thread-agnostic layouts only use contents of <see cref="LogEventInfo"/> for its output.
         /// </remarks>
-        internal bool ThreadAgnostic { get; private set; }
+        internal bool ThreadAgnostic { get; set; }
```

Comment:
I meant `protected internal`. Not sure if that could be set on the setter only. 

Its indeed and internal optimization (just makes the intention more clear).